compiler/Eta/DeSugar/Check.hs

```diff
@@ -56,7 +56,7 @@ The algorithm used is based on the paper:
 "GADTs Meet Their Match:
      Pattern-matching Warnings That Account for GADTs, Guards, and Laziness"
 
-http://people.cs.kuleuven.be/~george.karachalias/papers/p424-karachalias.pdf
+https://web.archive.org/web/20170706035904/https://people.cs.kuleuven.be/~george.karachalias/papers/p424-karachalias.pdf
```

Comment:
When a page wasn't reachable, I sometimes tried to find an archived version of the document. I didn't try for every page, and I have not done any checking on current `https` links -- that's beyond the scope of this PR.